src/components/editor/metaform-editor-right-drawer-feature.tsx

```diff
@@ -78,6 +78,13 @@ const MetaformEditorRightDrawerFeature: FC<Props> = ({
     updateSelected();
   }, [ sectionIndex, fieldIndex, pendingForm ]);
 
+  useEffect(() => {
+    setSelectMemberGroupEnabled(false);
+    setMemberGroupOptIndex(undefined);
+    setSelectedMemberGroupId(undefined);
+    setSelectedMemberGroupPermission(NOT_SELECTED);
+  }, [ sectionIndex, fieldIndex ]);
+
```

Comment:
Why is this necessary? Shouldn't checkIfMemberGroupsAreSelected handle the change of selected field already?
```
   /**
   * Empties member group settings
   */
  const emptyMemberGroupSettings = () => {
    setSelectMemberGroupEnabled(false);
    setMemberGroupOptIndex(undefined);
    setSelectedMemberGroupId(undefined);
    setSelectedMemberGroupPermission(NOT_SELECTED);
  };

  /**
   * Set member groups for selected field to the found first group
   */
  const checkIfMemberGroupsAreSelected = () => {
    if (sectionIndex === undefined || fieldIndex === undefined) {
      emptyMemberGroupSettings();
      return;
    }

    const updatedSelectedField = pendingForm.sections?.[sectionIndex].fields?.[fieldIndex];

    if (!updatedSelectedField || !MetaformUtils.fieldTypesAllowVisibility.includes(updatedSelectedField.type)) {
      emptyMemberGroupSettings();
      return;
    }

    const foundOptionIndex = updatedSelectedField.options?.findIndex(option => MetaformUtils.getOptionPermissionGroup(option) !== undefined);

    if (foundOptionIndex !== -1 && foundOptionIndex !== undefined) {
      const [ groupId, permission ] = MetaformUtils.getOptionPermissionGroup(updatedSelectedField.options![foundOptionIndex])!;

      setSelectMemberGroupEnabled(true);
      setSelectedMemberGroupId(groupId);
      setSelectedMemberGroupPermission(permission !== MemberGroupPermission.NOTIFY ? permission : NOT_SELECTED);
      setMemberGroupOptIndex(foundOptionIndex);
    } else {
      emptyMemberGroupSettings();
    }
  };

  useEffect(() => {
    updateSelected();
  }, [ sectionIndex, fieldIndex, pendingForm ]);

  useEffect(() => {
    checkIfMemberGroupsAreSelected();
  }, [ sectionIndex, fieldIndex, selectedField?.options?.length ]);
```
Maybe this one is better so you dont need an extra useeffect hook. And you could possibly replace the part where sets all the member group settings empty in removeAllPermissionGroups with emptyMemberGroupSettings

Reply:
That suggestion is not working for some reason. Renders wrong values sometimes.